lib/phoenix_live_view/test/live_view_test.ex

```diff
@@ -1028,6 +1034,125 @@ defmodule Phoenix.LiveViewTest do
     end
   end
 
+  @doc """
+  Open the default browser to display current HTML of `view_or_element`.
+
+  ## Examples
+
+      view
+      |> element("#term a:first-child()", "Increment")
+      |> open_browser()
+
+      assert view
+      |> form("#term", user: %{name: "hello"})
+      |> open_browser()
+      |> render_submit() =~ "Name updated"
+
+  """
+  def open_browser(view_or_element, open_fun \\ &open_with_system_cmd/1)
+
+  def open_browser(%View{} = view, open_fun) when is_function(open_fun, 1) do
+    view_html = render_tree(view, {proxy_topic(view), "render"})
```

Comment:
Is there a reason why can't call `render(view)` instead and that will take care of handling both views and elements? :)

Reply:
That's to pass floki's data structure to `maybe_wrap_html` where I'm injecting the `head` element if it's available on the root html. If I call `render(view)` I'd need to transform back from binary to list (floki's data structure), right?

Reply:
Oh, I see. So what you can do is that you can do the handling of `render_tree(view, {proxy_topic(view), "render"})` inside `render_tree`, to avoid the duplication in both render and open_browser.

Reply:
I hope that's what you meant [`866cc5b` (#1181)](https://github.com/phoenixframework/phoenix_live_view/pull/1181/commits/866cc5b002cf30ec57c65bf99e0077d7af2bb534#diff-39d73afd0f8def95e8eb610b5fa70619ff5a6755f2445aeea3d74d82a30f46a2R857) :)

Reply:
I meant something more like this:

```elixir
  def render(view_or_element) do
    view_or_element |> render_tree() |> DOM.to_html()
  end

  defp render_tree(%View{} = view) do
    render_tree(view, {proxy_topic(view), "render"})
  end
  defp render_tree(%Element{} = element) do
    render_tree(element, element)
  end
  defp render_tree(view_or_element, topic_or_element) do
    call(view_or_element, {:render_element, :find_element, topic_or_element})
  end
```

And you will call render_tree(view_or_element) in `open_browser`, removing the need for multiple clauses. :)